htdocs/core/class/commonobjectline.class.php

```diff
@@ -100,4 +100,255 @@ public function getLabelOfUnit($type = 'long')
 	// Currently we need function at end of file CommonObject for all object lines. Should find a way to avoid duplicate code.
 
 	// For the moment we use the extends on CommonObject until PHP min is 5.4 so use Traits.
+
```

Comment:
Why overriding the code of the showOptionals() already available ?
(Code seems really similar and introduce duplicate code)

Reply:
@eldy : It was a question during implementation. A parameters to say 'line' or 'form' or a different/extended method 
The code is actually similar, but after times it can became really different.
Display extrafields inside document line or inside card/form may have really different visual and rule.
1 - I prefers duplicated code for a time, as I bet in the future it will be different and then more easy to maintain
1' - if in the future "line.tpl"/"card.tpl (from module builder)" change it will be more easy to act only where it's called
2 - for maintainability, I prefer extended method because the number of line of the parent method is huge. In this case of showSeprator, I choose parameters because it have less long/complex code. Imagine if for each line that differs between this two methods actually, I use "If...else", number of lines increase, complexity increase, readability decrease
3 - extended method are recommanded is oriented object model to override behavior of the parent with keeping compatibility. With the tracks of earliest version of PHP (implicit interface), I seems to be a good option.

I can also counter-argument my own choice, but in the balance pro-con's this way of implementation win's for me. As often it's a point of view and it can be wrong. 

If you are not convinced, I can redo my work and put all inside parent method with a parameters.
My goal is to implement this feature into Dolibarr whatever the way ;-)